### PR TITLE
Added ability to edit text in a lowerthird

### PR DESCRIPTION
**Pull request by DomBelcher**
Adds a button to toggle editing in lower thirds

Turns heading and subheading into text fields where they can be edited.

Saves having to delete and redo them all the time

### CODE REVIEW

**Comment by benjaminkay93** [comment]
i was 50/50 with this, but it doesnt edit whats live on screen, however it feels a bit hit and miss with the edit button (am i just being daft?)

**Comment by benjaminkay93** [comment]
not editing live on screen was my only concern :P

**Comment by DomBelcher** [comment]
To edit the text live on screen I think you would need to keep track of what was showing to resend the show socket message, could be added in but a bit more complex. Can always just hit show again to change the text on screen

**Comment by benjaminkay93** [comment]
nooo live on screen edit is a bad thing aha,

**Comment by benjaminkay93** [comment]
the way it works rn is great, just the edit button seems abit ropy :P

**Comment by mstratford** [comment]
I like the feature, it's something that would have been useful in various (mostly non Roses specific) use cases. I'm also seeing the edit button only working sporadically.

Since you must click the green buttons to send the changes live, maybe something like changing the text of the green button to signify, "Update Left" or something could be done? The fact that the edit icon does not change (ie to a floppy "save" icon etc) seems like it could be better? :)

Somewhat off-topic, but it might be nice to disable/hide the ability to send the same lower third to multiple screen locations at a time?

**Comment by DomBelcher** [comment]
There's a few little style changes still to do in my mind, namely:
- change the edit button to a save button when in edit mode
- add some styling to the text boxes - maybe make the heading one bigger than the subheading

Other than those I think the feature works, I think other changes should be looked at in a separate pull request

**Comment by stephenrob** [comment]
Are you happy to make those minor style changes in this one before i review @DomBelcher ? I agree looking at the other changes separately.

**Comment by DomBelcher** [comment]
@stephenrob Yeah that's fine, I'll do them when I get the chance

File: public/admin/templates/lowerThirds.tmpl.html [hunk 1]
```diff
@@ -48,8 +52,8 @@ <h2 class="left aligned ui header">
         <div class="or"></div>
         <div class="ui positive button" ng-click="show('full', third)" >Full Width</div>
       </div>
-      <div class="ui icon red small button">
-        <i class="delete icon" ng-click='remove($index)'></i>
+      <div class="ui icon blue small button" ng-click='edit($index)'>
+        <i ng-class="third.edit ? 'save icon' : 'edit icon'"></i>
```
Comment (benjaminkay93) [inline public/admin/templates/lowerThirds.tmpl.html]: is it worth changing the color when the save button shows? maybe make it green

**Review by benjaminkay93 — COMMENTED** [review]
Can you add in a feature to block adding if the boxes are empty aswell please ? :) 
Apart from that its great :D

**Comment by DomBelcher** [comment]
@benjaminkay93 you mean if the edit inputs are empty, as opposed to the create ones?

**Comment by benjaminkay93** [comment]
screen shot 2018-04-13 at 00 18 45
Asin to stop that @DomBelcher :)

**Comment by DomBelcher** [comment]
@benjaminkay93 I can just make it so it prevents creating new lower thirds that are empty, and also editing them to be empty? Preventing creating empty ones isn't strictly related to the edit feature but seems very minor to open a separate PR for

**Comment by benjaminkay93** [comment]
those words are the words I was looking to say :P 
makes sense todo that if you can 👍

**Comment by DomBelcher** [comment]
Also, should it force having a heading, or is just a subheading acceptable?

**Comment by benjaminkay93** [comment]
screen shot 2018-04-13 at 00 29 14

I would say heading is required, sub heading is optional looking at how that displays :P

**Comment by DomBelcher** [comment]
All changes should be addressed as of https://github.com/LA1TV/Roses-CasparCG-Graphics/pull/98/commits/2a89b3e42852fdb64ca299e1c305d93beb3b4793

File: public/admin/templates/lowerThirds.tmpl.html [hunk 2]
```diff
@@ -48,8 +52,8 @@ <h2 class="left aligned ui header">
         <div class="or"></div>
         <div class="ui positive button" ng-click="show('full', third)" >Full Width</div>
       </div>
-      <div class="ui icon red small button">
-        <i class="delete icon" ng-click='remove($index)'></i>
+      <div  ng-class="third.edit ? 'green' : 'blue' "class="ui icon small button" ng-click='edit($index)'>
```
Comment (benjaminkay93) [inline public/admin/templates/lowerThirds.tmpl.html:55]: i never thought of using a trinary for this... good move 👍

**Review by benjaminkay93 — APPROVED** [review]
Looks really good, thank you @DomBelcher

**Pull request merged**